### PR TITLE
doc:update keycloak doc to add more information about the hasRole dir…

### PR DESCRIPTION
**Pull request by wei-lee**
…ective

### Description

Update the keycloak doc to add more information that is related to the `hasRole` directive

##### Checklist

- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](../doc/guides/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: doc/guides/keycloak_protection.md [hunk 1]
```diff
@@ -1,68 +1,118 @@
...
-const voyagerConfig = {
-  securityService: keycloakService
+type Query {
+  allPosts:[Post]!
+  getAuthor(id: ID!):Author!
 }
 
-const server = ApolloVoyagerServer(apolloConfig, voyagerConfig)
+type Mutation {
+  editPost:[Post]! @hasRole(role: ["editor", "admin"])
```
Comment (wtrocki) [inline doc/guides/keycloak_protection.md]: It seems like edit will not accept any input. To make it nicer we can just rename it to `upvotePost` that may not need any input.

**Comment by wtrocki** [comment]
Battle tested! Good to go!

File: doc/guides/keycloak_protection.md [hunk 2]
```diff
@@ -1,68 +1,118 @@
 ## Prerequisites:
 
-- You have provisioned Keycloak as described in [AeroGear Docs](https://docs.aerogear.org/aerogear/latest/identity-management.html).
+- There is a Keycloak service available.
+- You must first create a client for your application in the Keycloak Administration Console, and then click on the `Installation` tab, select `Keycloak OIDC JSON` for `Format` Option, and then click on `Download`. Save the downloaded `keycloak.json` file to your project.
 
 ## Setting up Keycloak protection in sync framework
```
Comment (darahayes) [inline doc/guides/keycloak_protection.md]: Minor thing: can we fix the capitalisation and naming? Should probably be something like `Setting up Keycloak Protection in the Voyager Framework`

File: doc/guides/keycloak_protection.md [hunk 3]
```diff
@@ -1,68 +1,118 @@
...
+    const app = express()
+    keycloakService.applyAuthMiddleware(app)
+    ```
+4. The last piece of work to do is telling Voyager server that the `keycloakService` above will be used as the security service:
+    ```javascript
+    const voyagerConfig = {
+      securityService: keycloakService
+    }
+    const server = ApolloVoyagerServer(apolloConfig, voyagerConfig)
+    ```
```
Comment (darahayes) [inline doc/guides/keycloak_protection.md]: Could we maybe have a new paragraph (single line) here after the given steps that links to the keycloak example server guide to help give the reader the bigger picture? Example:

```markdown
The [Keycloak Example Server Guide](https://github.com/aerogear/apollo-voyager-server/blob/master/doc/guides/examples.md#keycloak-example) has an example server based off the instructions above and shows all of the steps needed to get it running.
```

**Comment by wei-lee** [comment]
@darahayes your comments are addressed now. @wtrocki @darahayes can I get approval to merge this PR? Thanks.

**Pull request merged**